dids/diddht/internal/bep44/bep44.go

```diff
@@ -78,8 +80,24 @@ func (b *bep44Message) DecodePayload() ([]byte, error) {
 	return []byte(v), nil
 }
 
+// Encode encodes the BEP44 message into a byte slice, conforming to the Pkarr relay specification.
+func (msg *Message) Encode() ([]byte, error) {
+	// Construct the body of the request according to the Pkarr relay specification.
+	body := make([]byte, 0, len(msg.v)+72)
+	body = append(body, msg.sig...)
+	seq := uint64(msg.seq)
+
+	// Convert the sequence number to a big-endian byte array.
+	buf := make([]byte, 8) // uint64 is 8 bytes
+	binary.BigEndian.PutUint64(buf, seq)
+	body = append(body, buf...)
+	body = append(body, msg.v...)
+
+	return body, nil
+}
+
 // DecodeBEP44Message decodes the given byte slice into a BEP44 message.
-func DecodeBEP44Message(data []byte, b *bep44Message) error {
+func DecodeBEP44Message(data []byte, b *Message) error {
```

Comment:
I noticed [here](https://github.com/TBD54566975/web5-go/blob/dht-create/dids/diddht/diddht.go#L163) you're using `Marshal*` terminology  but are using `Encode/Decode*` here. any rationale behind the difference? Instinctively thought that `Marshal/Unmarshal` makes sense here as well

Reply:
I don't really have a strong opinion. I realized the inconsistency, but wasn't sure it's worth changing. 

On a "English is my 2nd language" note, I asked LLMs what the difference is:

```
Marshal/Unmarshal is more specific to object serialization and the transmission of complex data structures, often in the context of networking and distributed systems.

Encode/Decode is broader, referring to the conversion of data between different forms, including simple data types and complex objects, for various purposes like storage, transmission, and processing.
```

Reply:
Yeah I'm completely and entirely torn on this one so I will not cast a vote lol 

(I prefer encode/decode but the fact that Golang went with marshal/unmarshel with the json package really throws me for a loop)

Reply:
cool with whatever you decide @mihai-chiorean. primary goal for me was to understand what you were intending to convey when using Marshal/Unmarshal vs. Encode/Decode because i was unsure.

random thing that just came to mind is that it seems like the result of `Marshal` typically results in bytes.